### PR TITLE
Remove usige of relation regions from services

### PR DESCRIPTION
**Pull request by nessy**
- Related issue: https://myparcelcombv.atlassian.net/browse/MP-2275

### CODE REVIEW

File: src/MyParcelComApi.php [hunk 1]
```diff
@@ -296,18 +296,15 @@ public function getServices(ShipmentInterface $shipment = null, array $filters =
...
-            $shipment->getRecipientAddress()->getCountryCode(),
-            $shipment->getRecipientAddress()->getRegionCode()
-        )->get();
-
-        $url->addQuery($this->arrayToFilter([
-            'region_from' => reset($regionsFrom)->getId(),
-            'region_to'   => reset($regionsTo)->getId(),
+        $url->addQuery($this->arrayToFilters([
+            'address_from' => [
+                'country_code'=>$shipment->getSenderAddress()->getCountryCode(),
```
Comment (NickVries) [inline src/MyParcelComApi.php]: Autoformatting should fix indentation here.

File: src/MyParcelComApi.php [hunk 2]
```diff
@@ -817,16 +814,34 @@ protected function getResourceUri($resourceType, $id = null)
      * @param array $array
      * @return array
      */
-    private function arrayToFilter(array $array)
+    private function arrayToFilters(array $array)
```
Comment (NickVries) [inline src/MyParcelComApi.php]: Can you try to rewrite this and the next method? I have no idea what's going on. Do we really need two methods?

File: src/MyParcelComApi.php [hunk 3]
```diff
@@ -817,16 +814,34 @@ protected function getResourceUri($resourceType, $id = null)
...
     }
 
+    /**
+     * Converts given array to a filter string for the query params.
+     *
+     * @param array $keys
+     * @param $values
+     * @return array
+     */
+    private function arrayToFilter($keys, $values)
```
Comment (NickVries) [inline src/MyParcelComApi.php]: What are these `keys` and `values` that we're expecting in this method? I can't really follow what's happening.

File: src/Resources/ResourceFactory.php [hunk 4]
```diff
@@ -195,6 +196,54 @@ protected function serviceFactory(array &$properties)
             unset($properties['attributes']['transit_time']['max']);
         }
 
+        if (isset($properties['attributes']['regions_from'][0])) {
+            $regionFromAttribute = $properties['attributes']['regions_from'][0];
+
+            $countryCode = $regionFromAttribute['country_code'];
+            $regionCode = $regionFromAttribute['region_code'];
```
Comment (NickVries) [inline src/Resources/ResourceFactory.php]: Region code might not always be set. If it isn't we should query the API with `region_code` `null`. 
```suggestion
            $regionCode = $regionFromAttribute['region_code'] ?? null;
```

File: src/Resources/ResourceFactory.php [hunk 5]
```diff
@@ -195,6 +196,54 @@ protected function serviceFactory(array &$properties)
...
+
+            /** @var RegionInterface $regionFrom */
+            $regionFrom = $this->api->getRegions([
+                    'country_code' => $countryCode,
+                    'region_code'  => $regionCode,
+                ]
```
Comment (NickVries) [inline src/Resources/ResourceFactory.php]: The closing method paranthesis `)` can be on the same line as the closing array bracket.

```suggestion
                ])->get()[0];
```

File: src/Resources/ResourceFactory.php [hunk 6]
```diff
@@ -195,6 +196,54 @@ protected function serviceFactory(array &$properties)
...
+            )->get()[0];
+
+            if(!$regionFrom){
+                throw new ResourceNotFoundException(
+                    'No region found for country code '.$countryCode.' and region code '. $regionCode
```
Comment (NickVries) [inline src/Resources/ResourceFactory.php]: If `$regionCode` is `null`, the sentence will end in: "and region code ". Maybe we can stringify it if it is `null` to indicate that there was no region code.

File: src/Resources/ResourceFactory.php [hunk 7]
```diff
@@ -195,6 +196,54 @@ protected function serviceFactory(array &$properties)
...
+            $regionCode = $regionFromAttribute['region_code'];
+
+            /** @var RegionInterface $regionFrom */
+            $regionFrom = $this->api->getRegions([
+                    'country_code' => $countryCode,
+                    'region_code'  => $regionCode,
+                ]
+            )->get()[0];
+
+            if(!$regionFrom){
```
Comment (NickVries) [inline src/Resources/ResourceFactory.php]: Autoformatting should fix the spacing here.

File: src/Resources/ResourceFactory.php [hunk 8]
```diff
@@ -195,6 +196,54 @@ protected function serviceFactory(array &$properties)
...
+            $regionToAttribute = $properties['attributes']['regions_to'][0];
+
+            $countryCode = $regionToAttribute['country_code'];
+            $regionCode = $regionToAttribute['region_code'];
+
+            /** @var RegionInterface $regionTo */
+            $regionTo = $this->api->getRegions([
+                    'country_code' => $countryCode,
+                    'region_code'  => $regionCode,
+                ]
```
Comment (NickVries) [inline src/Resources/ResourceFactory.php]: Same as above, the end of the method call can be on the same line as the closing array bracket.

File: tests/Unit/ServiceTest.php [hunk 9]
```diff
@@ -216,6 +216,18 @@ public function testJsonSerialize()
                 'delivery_days'   => [
                     'Monday',
                 ],
+                'regions_from' => [
```
Comment (NickVries) [inline tests/Unit/ServiceTest.php]: Why did you add this here? Shouldn't region relationships have been removed all together?

File: src/MyParcelComApi.php [hunk 10]
```diff
@@ -817,16 +814,35 @@ protected function getResourceUri($resourceType, $id = null)
...
+     * Converts given array to a filter string for the query params.
+     *
+     * @param array $keys
+     * @param       $values
+     * @return array
+     */
+    private function arrayToFilter($keys, $values)
+    {
+        if (!is_array($values)) {
+            return ['[' . implode('][', $keys) . ']', $values];
```
Comment (M4tini) [inline src/MyParcelComApi.php]: `$keys)` ?
